includes/notes/class-wc-payments-notes-set-up-stripelink.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Set up Stripe Link note for WooCommerce inbox.
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Payments_Notes_Setup_StripeLink
+ */
+class WC_Payments_Notes_Set_Up_StripeLink {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-payments-notes-set-up-stripe-link';
+
+	/**
+	 * CTA button link
+	 */
+	const NOTE_DOCUMENTATION_URL = 'https://woocommerce.com/document/payments/woocommerce-payments-stripe-link/';
+
+	/**
+	 * The payment gateway service instance.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private static $gateway;
+
+	/**
+	 * Checks if a note can and should be added.
+	 *
+	 * @return bool
+	 */
+	public static function should_display_note() {
+		$payment_method_statuses = self::$gateway->get_upe_enabled_payment_method_statuses();
```

Comment:
And what happens if the gateway is not yet injected?

Reply:
Not applicable anymore.